packages/react/gulpfile.js

```diff
@@ -385,8 +389,102 @@ function cleanIconDir() {
   ]);
 }
 
+const typedSources = [
+  ...sources,
+
+  // @todo This file probably requires a separate transformation using JSCodeShift.
+  '!src/index.js',
+];
+
+const tsIcons = [
+  "types/components/base/Icon/*.tsx"
+];
+
+const tsDeclarationSources = [
+  'types/**/*.tsx',
+];
+
+function generateTsIcons() {
+  return run('svgr --out-dir ./types/components/base/Icon ./src/components/base/Icon/assets --config-file=./.svgrrc-ts.js')()
+}
+
+function ignoreTsCheckOnIcons() {
+  return src(tsIcons, {base: 'types'})
+    .pipe(prependText('// @ts-nocheck'))
+    .pipe(dest('types'))
+}
+
+function createTsCopy() {
+  return src(typedSources)
+    .pipe(rename((p) => {
+      // eslint-disable-next-line no-param-reassign
+      p.extname = '.tsx'
+    }))
+    .pipe(prependText('// @ts-nocheck'))
+    .pipe(dest('types'))
+}
+
+function convertPropTypesToTs(cb) {
+  // We have to use `exec` here in order to use the transformation file written in TS.
+  exec(`./node_modules/.bin/jscodeshift -t "scripts/transform.ts" --extensions=tsx --fail-on-error "types"`, (error, stdout, stderr) => {
+    if (stdout) {
+      console.log(stdout);
+    }
+    if (stderr) {
+      console.error(stderr);
+    }
+
+    cb(error);
+  });
+}
+
+function convertTsToDeclarations() {
+  const tsProject = ts.createProject('tsconfig.json')
+  return src(tsDeclarationSources)
+    .pipe(rename((p) => {
+      const splitPath = p.dirname.split('/');
+      // eslint-disable-next-line no-param-reassign
+      p.dirname = splitPath[splitPath.length - 1];
+    }))
+    .pipe(tsProject())
+    .dts
+    .pipe(
+      // @todo Is there a better way to do it? Some TS compiler options maybe?
+      replace(/(\s+from\s+['"])(Mayflower[a-z]+)\//ig, '$1../')
+    )
+    .pipe(dest('dist'))
+}
+
+const generateTsDeclarations = series(
```

Comment:
what components are excluded as of this PR? 
Also, are defaultProps handled by the transformation?

Reply:
e.g. I noticed that CompoundSlider has defaultProps that doesn't seem to reflect in types

Reply:
defaultProps are not handled, but all components have types declarations